custom_components/waste_collection_schedule/waste_collection_schedule/source/ccc_govt_nz.py

```diff
@@ -1,77 +1,78 @@
 import datetime
 
-import requests
 from waste_collection_schedule import Collection
-
-# Include work around for SSL UNSAFE_LEGACY_RENEGOTIATION_DISABLED error
 from waste_collection_schedule.service.SSLError import get_legacy_session
 
 
 TITLE = "Christchurch City Council"
-DESCRIPTION = "Source for Christchurch City Council."
-URL = "https://ccc.govt.nz"
-TEST_CASES = {"53 Hereford Street": {"address": "53 Hereford Street"}}
+DESCRIPTION = "Source for Christchurch City Council"
+URL = "https://ccc.govt.nz/services/rubbish-and-recycling/collections"
+TEST_CASES = {
+    "53 Hereford Street": {"address": "53 Hereford Street"}
+}
+
+ICON_MAP = {
+    "Garbage": "mdi:trash-can",
+    "Recycle": "mdi:recycle",
+    "Organic": "mdi:leaf",
+}
 
 
 class Source:
-    def __init__(self, address):
+    def __init__(self, address:str):
         self._address = address
 
-    def fetch(self):
-
-        s = get_legacy_session()
+    def fetch(self) -> list[Collection]:
 
         entries = []
 
         # Find the Rating Unit ID by the physical address
         # While a property may have more than one address, bins are allocated by each Rating Unit
-        addressQuery = {
-            "q": self._address,
-            "status": "current",
-            "crs": "epsg:4326",
-            "limit": 1,
-        }
-
-        r = s.get("https://opendata.ccc.govt.nz/CCCSearch/rest/address/suggest",
-            params=addressQuery,
-            # verify=False,
+        r = get_legacy_session().get("https://opendata.ccc.govt.nz/CCCSearch/rest/address/suggest",
```

Comment:
Any Reason you use the legacy session, I don't get any SSL errors running with normal SSL validation